packages/@react-aria/dnd/src/useDraggableItem.ts

```diff
@@ -48,19 +80,63 @@ export function useDraggableItem(props: DraggableItemProps, state: DraggableColl
 
   let item = state.collection.getItem(props.key);
   let numKeysForDrag = state.getKeysForDrag(props.key).size;
-  let isSelected = state.selectionManager.isSelected(props.key);
-  let message: string;
-  if (isSelected && numKeysForDrag > 1) {
-    message = stringFormatter.format('dragSelectedItems', {count: numKeysForDrag});
+  let isSelected = numKeysForDrag > 1 && state.selectionManager.isSelected(props.key);
+  let dragButtonLabel: string;
+  let description: string;
+
+  // Override description to include selected item count.
+  let modality = useDragModality();
+  if (!props.hasDragButton) {
+    let msg = MESSAGES[modality][isSelected ? 'selected' : 'notSelected'];
+    if (props.hasAction && modality === 'keyboard') {
+      msg += 'Alt';
+    }
+
+    if (isSelected) {
+      description = stringFormatter.format(msg, {count: numKeysForDrag});
+    } else {
+      description = stringFormatter.format(msg);
+    }
   } else {
-    message = stringFormatter.format('dragItem', {itemText: item?.textValue ?? ''});
+    if (isSelected) {
+      dragButtonLabel = stringFormatter.format('dragSelectedItems', {count: numKeysForDrag});
+    } else {
+      dragButtonLabel = stringFormatter.format('dragItem', {itemText: item?.textValue ?? ''});
+    }
+  }
+
+  let descriptionProps = useDescription(description);
+  if (description) {
+    Object.assign(dragProps, descriptionProps);
+  }
+
+  if (!props.hasDragButton && props.hasAction) {
+    let {onKeyDownCapture, onKeyUpCapture} = dragProps;
+    if (modality === 'touch') {
+      // Remove long press description if an action is present, because in that case long pressing selects the item.
+      delete dragProps['aria-describedby'];
+    }
+
+    // Require Alt key if there is a conflicting action.
+    dragProps.onKeyDownCapture = e => {
+      if (e.altKey) {
+        onKeyDownCapture(e);
+      }
+    };
+
+    dragProps.onKeyUpCapture = e => {
+      if (e.altKey) {
+        onKeyUpCapture(e);
+      }
+    };
   }
 
   return {
-    dragProps,
+    dragProps: isDisabled ? {} : dragProps,
     dragButtonProps: {
       ...dragButtonProps,
-      'aria-label': message
+      isDisabled,
+      'aria-label': dragButtonLabel
```

Comment:
Won't disabled items erroneously get the `dragItem`/`dragSelectedItems` messaging here?

Reply:
Ah, I guess the user would/should not render the drag handle for the disabled drag item

Reply:
And even so, it would read as disabled so I think it's ok